src/main/java/Duke.java

```diff
@@ -1,10 +1,208 @@
+        } catch (NumberFormatException e) {
+            System.out.println("Please specify the task you would like to mark as done in integer format.");
+        } catch (NullPointerException e) {
+            System.out.println("Task not found. Please reenter command with valid task number.");
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Task not found. Please reenter command with valid task number.");
+        }
+    }
+
+    //Adds task at end of other tasks
+    public static void addTask(String userInput) {
+
+        String taskType = "", taskName = "";
+        try {
+            taskType = userInput.split(" ")[0];
+            taskName = userInput.split(" ", 2)[1].split(" /")[0];
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("OOPS!!! The description of " + taskType + " cannot be empty.");
+            return;
+        }
+
+        switch (taskType) {
+        case "todo":
+            tasks.add(new Todo(taskName));
+            break;
+        case "deadline":
+            try {
+                String deadline = userInput.split("/by ")[1];
+                tasks.add(new Deadline(taskName, deadline));
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("Missing deadline or deadline not specified in correct format.");
+                return;
+            }
+            break;
+        case "event":
+            try {
+                String time = userInput.split("/at ")[1];
+                tasks.add(new Event(taskName, time));
+                break;
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("Missing event time or time not specified in correct format.");
+                return;
+            }
+        }
+        System.out.format("Got it. I've added this task: %n");
+        System.out.format("%s%n", tasks.get(Task.getNumberOfTasks() - 1).toString());
+        System.out.format("Now you have %d tasks on the list.%n", Task.getNumberOfTasks());
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
+        String logo = "____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
+
+        //Prints welcome message on start
         System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+
+        //Load previous data from file
+        readFile();
+
+        //Processes user commands
+        String userInput = in.nextLine();
+
+        while (!userInput.equals("bye")) {
+            try {
+                String commandType = userInput.split(" ")[0];
+                switch (commandType) {
+                case "list":
```

Comment:
Consider using named constants in place of the command words.

Reply:
Thanks for the suggestion. I always thought named constants were only to avoid magic numbers and constants that are unclear. What's the purpose of them here or is it just for convention?

Reply:
the command words are also _sort of_ magic literals, no?